src/test/codewhisperer/tracker/codewhispererCodeCoverageTracker.test.ts

```diff
@@ -62,18 +191,20 @@ describe('codewhispererCodecoverageTracker', function () {
             resetCodeWhispererGlobalVariables()
             CodeWhispererCodeCoverageTracker.instances.delete(language)
         })
-
+        afterEach(function () {
+            sinon.restore()
+        })
         it(' emits codecoverage telemetry ', function () {
-            const sample = "print('Hello World!)"
             const tracker = CodeWhispererCodeCoverageTracker.getTracker(language, mockGlobalStorage)
             const assertTelemetry = assertTelemetryCurried('codewhisperer_codePercentage')
-            tracker.setAcceptedTokens(sample)
-            tracker.emitCodeWhispererCodeContribution()
+            tracker?.addAcceptedTokens(`test.py`, { range: new vscode.Range(0, 0, 0, 7), text: `print()`, accepted: 7 })
+            tracker?.addTotalTokens(`test.py`, 100)
+            tracker?.emitCodeWhispererCodeContribution()
             assertTelemetry({
```

Comment:
Having more tests using `assertTelemetry` would be nice if possible. Like I want to see how different combinations of user actions produce different results.

Reply:
New test cases have been added.